### PR TITLE
Improve docs

### PR DESCRIPTION
**Pull request by ironcladlou**
/cc @openshift/sig-network-edge @smarterclayton @derekwaynecarr @bmcelvee

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,12 @@
 # Cluster Ingress Operator
 
-The Cluster Ingress Operator manages highly available network ingress for [OpenShift](https://openshift.io/). Simplicity is a design goal: minimal configuration should be required to materialize a highly available ingress solution in both cloud and bare metal environments.
+Cluster Ingress Operator deploys and manages [HAProxy](http://www.haproxy.org) to provide highly available network ingress in [OpenShift](https://openshift.io/), enabling both OpenShift [Routes](https://docs.okd.io/latest/rest_api/apis-route.openshift.io/v1.Route.html) and Kubernetes [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/).
```
Comment (Miciah) [inline README.md]: s/Ingress/Ingresses/

File: README.md [hunk 2]
```diff
@@ -1,7 +1,12 @@
...
 
-## Development
+The operator tries to be useful out of the box by creating a working default deployment based on the cluster's configuration.
 
-See [HACKING](HACKING.md) for development topics.
+* On cloud platforms, ingress is exposed externally using Kubernetes [LoadBalancer Services](https://kubernetes.io/docs/concepts/services-networking/#loadbalancer).
+* On Amazon AWS, wildcard DNS for the ingress domain is automatically managed.
```
Comment (Miciah) [inline README.md]: How about, "the operator publishes a wildcard DNS record based on the [cluster ingress domain](https://github.com/openshift/api/blob/master/config/v1/types_ingress.go) and pointing to the service's load-balancer"?

File: HACKING.md [hunk 3]
```diff
@@ -1,94 +1,48 @@
...
-$ hack/uninstall.sh
-
-# 2. Build and push a new cluster-ingress-operator image.
-$ REPO=docker.io/username/origin-cluster-ingress-operator make release-local
-
-# 3. Run `oc apply` as instructed to install the locally-built operator.
-
-# 4. Run integration tests against the cluster.
-$ CLUSTER_NAME=your-cluster-name make test-integration
+$ make test-e2e
```
Comment (Miciah) [inline HACKING.md]: Should we say anything about running the origin tests?

File: README.md (hunk 1 above)
Comment (ironcladlou) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (ironcladlou) [reply README.md]: Fixed anyway in advance of the PR since merge is imminent

File: HACKING.md (hunk 3 above)
Comment (ironcladlou) [reply HACKING.md]: Maybe -- I'd like to think more about it and do some followup improvements here and in the DNS operator doc with other test info and also some alternative development workflows

**Pull request merged**